src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -0,0 +1,1003 @@
+
+  /* a wrapped (see ...) on-complete closure for readily available RR picks */
+  grpc_closure wrapped_on_complete;
+
+  /* arguments for the wrapped_on_complete closure */
+  wrapped_rr_closure_arg wc_arg;
+};
+
+static void rr_handover(grpc_exec_ctx *exec_ctx, glb_lb_policy *p,
+                        grpc_error *error);
+static void rr_connectivity_changed(grpc_exec_ctx *exec_ctx, void *arg,
+                                    grpc_error *error) {
+  rr_connectivity_data *rrcd = arg;
+  glb_lb_policy *p = rrcd->p;
+  if (rrcd->state == GRPC_CHANNEL_SHUTDOWN) {
+    if (p->serverlist != NULL) {
+      /* a RR policy is shutting down but there's a serverlist available ->
+       * perform a handover */
+      rr_handover(exec_ctx, p, error);
+    } else {
+      /* shutting down and no new serverlist available. Bail out. */
+      gpr_free(rrcd);
+    }
+  } else {
+    if (error == GRPC_ERROR_NONE) {
+      /* RR not shutting down. Mimic the RR's policy state */
+      grpc_connectivity_state_set(exec_ctx, &p->state_tracker, rrcd->state,
+                                  error, "rr_connectivity_changed");
+      /* resubscribe */
+      grpc_lb_policy_notify_on_state_change(exec_ctx, p->rr_policy,
+                                            &rrcd->state, &rrcd->on_change);
+    } else { /* error */
+      gpr_free(rrcd);
+    }
+  }
+  GRPC_ERROR_UNREF(error);
+}
+
+static void add_pending_pick(pending_pick **root, grpc_polling_entity *pollent,
+                             grpc_metadata_batch *initial_metadata,
+                             uint32_t initial_metadata_flags,
+                             grpc_connected_subchannel **target,
+                             grpc_closure *on_complete) {
+  pending_pick *pp = gpr_malloc(sizeof(*pp));
+  memset(pp, 0, sizeof(pending_pick));
+  memset(&pp->wrapped_on_complete_arg, 0, sizeof(wrapped_rr_closure_arg));
+  pp->next = *root;
+  pp->pollent = pollent;
+  pp->target = target;
+  pp->initial_metadata = initial_metadata;
+  pp->initial_metadata_flags = initial_metadata_flags;
+  pp->wrapped_on_complete_arg.wrapped_closure = on_complete;
+  grpc_closure_init(&pp->wrapped_on_complete, wrapped_rr_closure,
+                    &pp->wrapped_on_complete_arg);
+  *root = pp;
+}
+
+static void add_pending_ping(pending_ping **root, grpc_closure *notify) {
+  pending_ping *pping = gpr_malloc(sizeof(*pping));
+  memset(pping, 0, sizeof(pending_ping));
+  memset(&pping->wrapped_notify_arg, 0, sizeof(wrapped_rr_closure_arg));
+  pping->next = *root;
+  grpc_closure_init(&pping->wrapped_notify, wrapped_rr_closure,
+                    &pping->wrapped_notify_arg);
+  pping->wrapped_notify_arg.wrapped_closure = notify;
+  *root = pping;
+}
+
+static void lb_client_data_destroy(lb_client_data *lbcd);
+
+static void md_sent_cb(grpc_exec_ctx *exec_ctx, void *arg, grpc_error *error) {
+  lb_client_data *lbcd = arg;
+  GPR_ASSERT(lbcd->lb_call);
+  grpc_call_error call_error;
```

Comment:
yeah... this is a testament to how long this has been under development. We couldn't do that when all the core was C89.

Fixed in some obvious places. I'm sure there are still some others that could be declared-initialized in C99 style.